device_backends/mmio/include/DeviceFile.h

```diff
@@ -11,18 +11,22 @@ namespace ChimeraTK {
   // Simple RAII wrapper for a device file
   class DeviceFile : private boost::noncopyable {
    private:
-    std::string _strerror(const std::string& msg) const;
-    const std::string _path;
-    int _fd;
+    [[nodiscard]] std::string _strerror(const std::string& msg) const;
+    std::string _path{};
+    int _fd{-1};
+    int _savedErrno{0};
+    bool _fdOwner{true};
 
    public:
     DeviceFile() = delete;
-    DeviceFile(const std::string& filePath, int flags);
-    DeviceFile(DeviceFile&& d);
+    DeviceFile(std::string filePath, int flags);
+    DeviceFile(int fd, bool takeFdOwnership = true);
+    DeviceFile(DeviceFile&& d) noexcept;
     virtual ~DeviceFile();
 
-    operator int() const;
-    std::string name() const;
+    explicit operator int() const;
```

Comment:
Please add some documentation. What the conversion to int does for a DeviceFile is really not obvious...

Reply:
Removed. It was missing the implementation anyway because I decided for the explicit `fd()` call later.